### PR TITLE
Remove list initialization for prefix_scan_sum

### PR DESCRIPTION
**Pull request by TedThemistokleous**
This seems to cause an error as seen in : https://ontrack-internal.amd.com/browse/SWDEV-443633

Issue with using list initialization for inputs in the compute shape function of prefix_scan_sum.

related to #2827

### CODE REVIEW

**Comment by pfultz2** [comment]
I think it would be better to make the `shape(const std::vector<shape>& subs)` explicit, that might fix the issue which seems due to implicit conversions.

File: src/shape.cpp [hunk 1]
```diff
@@ -95,7 +95,9 @@ struct shape_impl
         }
     }
 
-    shape_impl(const std::vector<shape>& subs) : m_type(shape::tuple_type), m_shapes(subs) {}
+    explicit shape_impl(const std::vector<shape>& subs) : m_type(shape::tuple_type), m_shapes(subs)
```
Comment (pfultz2) [inline src/shape.cpp:98]: This shouldn't need explicit here, this is just a private class.

Comment (TedThemistokleous) [reply src/shape.cpp:98]: Sure I'll change.

Comment (TedThemistokleous) [reply src/shape.cpp]: Done

File: src/targets/gpu/include/migraphx/gpu/prefix_scan_sum.hpp [hunk 2]
```diff
@@ -54,7 +54,7 @@ struct hip_prefix_scan_sum : oper<hip_prefix_scan_sum>
 
     shape compute_shape(const std::vector<shape>& inputs) const
     {
-        std::vector<shape> in_shapes{inputs};
+        std::vector<shape> in_shapes(inputs);
```
Comment (umangyadav) [inline src/targets/gpu/include/migraphx/gpu/prefix_scan_sum.hpp:57]: is this necessary change now that constructor is marked as explicit ?

Comment (TedThemistokleous) [reply src/targets/gpu/include/migraphx/gpu/prefix_scan_sum.hpp]: Changed it back to braced construction

**Review by pfultz2 — APPROVED** [review]

**Review by umangyadav — APPROVED** [review]

**Pull request merged**